test_msvc_examples failure on SPARC

When running the test suite on SPARC, we're getting the following test failure:

```pytb
$ python -m pytest
========================================================= test session starts =========================================================
platform linux -- Python 3.11.9, pytest-8.2.2, pluggy-1.5.0
rootdir: /home/mgorny/crc32c
collected 13 items                                                                                                                    

test/test_crc32c.py .F...........                                                                                               [100%]

============================================================== FAILURES ===============================================================
_____________________________________________________ TestMisc.test_msvc_examples _____________________________________________________

self = <test.test_crc32c.TestMisc testMethod=test_msvc_examples>

    def test_msvc_examples(self):
        # Examples taken from MSVC's online examples.
        # Values are not xor'd in the examples though, so we do it here
        max32 = 0xffffffff
        def assert_msvc_vals(b, crc, expected_crc):
            self.assertEqual(expected_crc ^ max32, crc32c.crc32c(b, crc ^ max32))
        assert_msvc_vals(uchar_as_bytes(100), 1, 1412925310)
        assert_msvc_vals(ushort_as_bytes(1000), 1, 3870914500)
        assert_msvc_vals(uint_as_bytes(50000), 1, 971731851)
>       assert_msvc_vals(ulonglong_as_bytes(0x88889999EEEE3333), 0x5555AAAA, 0x16F57621)

test/test_crc32c.py:96: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test/test_crc32c.py:92: in assert_msvc_vals
    self.assertEqual(expected_crc ^ max32, crc32c.crc32c(b, crc ^ max32))
E   AssertionError: 3909781982 != 2426753527
======================================================= short test summary info =======================================================
FAILED test/test_crc32c.py::TestMisc::test_msvc_examples - AssertionError: 3909781982 != 2426753527
==================================================== 1 failed, 12 passed in 0.19s =====================================================
```

Our initial thought would be that it's an endianness problem, but apparently the same tests passes on big-endian PPC64.

--- a/crc32c_sw.c
+++ b/crc32c_sw.c
@@ -490,7 +490,7 @@ const uint32_t crc_tableil8_o88[256] =
 
 uint32_t _crc32c_sw_slicing_by_8(uint32_t crc, unsigned const char* data, unsigned long length)
 {
-	const char* p_buf = (const char*) data;
+	unsigned const char* p_buf = data;
 	size_t initial_bytes = (sizeof(uint32_t) - (intptr_t)p_buf) & (sizeof(uint32_t) - 1);
 	size_t li;
 	size_t running_length;

